File: Zadania/2025/Zadanie-6/Java/Gildia.java
package Java;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class Gildia {
    String nazwaGildi;
    String liderGildi;
    List<Druzyna> listaDruzyn;

    Gildia(String nazwaGildi, String liderGildi, List<Druzyna> listaDruzyn) {
        this.nazwaGildi = nazwaGildi;
        this.liderGildi = liderGildi;
        this.listaDruzyn = new ArrayList<Druzyna>();
    }

    public void dodajDruzyne(Druzyna druzyna) {
        this.listaDruzyn.add(druzyna);
    }

    public Wojownik najsilniejszyWojownik() {
        Wojownik najsilniejszy = null;
        for (Druzyna druzyna : listaDruzyn) {
            for (Wojownik wojownik : druzyna.getCzlonkowie()) {
                if (najsilniejszy == null || wojownik.getSila() > najsilniejszy.getSila()) {
                    najsilniejszy = wojownik;
                }
            }
        }
        return najsilniejszy;
    }

    public Wojownik najmocniejszyPoziom() {
        Wojownik najwyzszyPoziom = null;

        for (Druzyna druzyna : listaDruzyn) {
            for (Wojownik wojownik : druzyna.getCzlonkowie()) {
                if (najwyzszyPoziom == null || wojownik.getPoziom() > najwyzszyPoziom.getPoziom()) {
                    najwyzszyPoziom = wojownik;
                }
            }
        }
        return najwyzszyPoziom;
    }

    public static void main(String[] args) {
        Gildia gildia = new Gildia("Cienie Valhalli", "Ragnar", null);
        String[] imiona = { "Thorin", "Eldrin", "Gorash", "Liriel", "Azog" };
        String[] rasy = { "Krasnolud", "Elf", "Ork", "Czlowiek" };
        String[] nazwyDruzyn = { "Wilki Północy", "Strażnicy Ognia", "Mroczne Ostrza" };
        Random random = new Random();
        for (String nazwa : nazwyDruzyn) {
            Druzyna druzyna = new Druzyna(nazwa, null);
            for (int i = 0; i < 5; i++) {
                Wojownik wojownik = new Wojownik(imiona[random.nextInt(imiona.length)],
                        rasy[random.nextInt(rasy.length)],
                        random.nextInt(100) + 1,
                        1 + random.nextDouble() * 9);
                druzyna.dodajWojownika(wojownik);  
            }
            gildia.dodajDruzyne(druzyna);
        }
        System.out.println("Najsilniejszy wojownik: " + gildia.najsilniejszyWojownik());
        System.out.println("Wojownik z najwyższym poziomem: " + gildia.najmocniejszyPoziom());
    }
}

File: Zadania/2025/Zadanie-6/Python/Gildia.py
from Wojownik import Wojownik
from Druzyna import Druzyna
import random


class Gildia:
    def __init__(self , nazwaGildi , mistrzGildi):
        self.nazwaGildi = nazwaGildi
        self.mistrzGildi = mistrzGildi
        self.druzyny = [Druzyna]

    def dodajDruzyne(self , druzyna):
        self.druzyny.append(druzyna)

    def najsilnieszyWoj(self):
        najsilniejszy = 0
        for druzyna in self.druzyny:
            for wojownik in druzyna.czlonkowie:
                if najsilniejszy is 0 or wojownik.sila > najsilniejszy.sila:
                    najsilniejszy = wojownik
        return najsilniejszy
    
    def najwiekszyLvl(self):
        najwiekszyLvl = 0
        for druzyna in self.druzyny:
            for wojownik in druzyna.czlonkowie:
                if najwiekszyLvl is 0 or wojownik.level > najwiekszyLvl.level:
                    najwiekszyLvl = wojownik
        return najwiekszyLvl
    
gildia = Gildia("Cienie Valhalli" , "Ragnar")

imiona = ["Thorin", "Eldrin", "Gorash", "Liriel", "Azog"]
rasy = ["Elf", "Ork", "Człowiek", "Krasnolud"]
nazwy_druzyn = ["Wilki Północy", "Strażnicy Ognia", "Mroczne Ostrza"]

for nazwa in nazwy_druzyn:
    druzyna = Druzyna(nazwa)
    for i in range(5):
        wojownik =Wojownik(
            random.choice(imiona),
            random.choice(rasy),
            random.randint(1,100),
            round(random.uniform(1.0, 10.0), 2)
            )
        druzyna.dodajWojownika(wojownik)
    gildia.dodajDruzyne(druzyna)
print(f"Najsilniejszy wojownik: {gildia.najsilnieszyWoj()}")
print(f"Wojownik z najwyższym poziomem: {gildia.najwiekszyLvl()}")



